File: NegrobovVAAS-21-05lab2/NegrobovVAAS-21-05lab2/NegrobovSSSPresident.cpp
#include "pch.h"
#include "NegrobovSSSPresident.h"

const int min_experience = 0;
const int max_experience = 10;

IMPLEMENT_SERIAL(NegrobovSSSPresident, NegrobovSSSMember, VERSIONABLE_SCHEMA | 0)

void NegrobovSSSPresident::input() {

    NegrobovSSSMember::input();
    std::cout << "Input experience(min: 0; max: 10): ";
    experience = get_num_value(0, 10);
}


void NegrobovSSSPresident::output() {
    std::cout << "=======================================" << std::endl;
    std::cout << "President" << std::endl;
    NegrobovSSSMember::output();
    std::cout << "Experience: "
        << experience << std::endl
        << "=======================================" << std::endl;
}


void NegrobovSSSPresident::Serialize(CArchive& ar) {

    NegrobovSSSMember::Serialize(ar);

    if (ar.IsStoring()) {
        ar << experience;
    }
    else {
        ar >> experience;
    }

}

File: NegrobovVALab5/NegrobovSSSPresident.cs
using System;

namespace NegrobovVALab5
{
    [Serializable]
    public class NegrobovSSSPresident : NegrobovSSSMember
    {
        uint experience;

        public override output()
        {
            Console.WriteLine("-----------------------------------------");
            Console.WriteLine("President");
            base.output();
            Console.WriteLine("experience: ", experience);
            Console.WriteLine("-----------------------------------------");

        }

        public override input()
        {
            base.input();
            Console.Write("Input experience: ");
            experience = Convert.ToUInt32(Console.ReadLine("Input "));
        }
    }

}
